From: olh@suse.de
Subject: fix use after potential free in scsi_cd_put

# ChangeSet
#   2005/01/17 12:08:43-06:00 jejb@mulgrave.(none) 
#   fix use after potential free in scsi_cd_put
#   
#   kref_put could free the cd structure.  We need a copy
#   of cd->device to do a scsi_device_put after the
#   kref_put.
#   
#   Signed-off-by: James Bottomley <James.Bottomley@SteelEye.com>
# 
# drivers/scsi/sr.c
#   2005/01/17 12:08:00-06:00 jejb@mulgrave.(none) +3 -1
#   fix use after potential free in scsi_cd_put
# 
diff -Nru a/drivers/scsi/sr.c b/drivers/scsi/sr.c
--- a/drivers/scsi/sr.c	2005-01-23 22:28:26 -08:00
+++ b/drivers/scsi/sr.c	2005-01-23 22:28:26 -08:00
@@ -152,9 +152,11 @@
 
 static inline void scsi_cd_put(struct scsi_cd *cd)
 {
+	struct scsi_device *sdev = cd->device;
+
 	down(&sr_ref_sem);
 	kref_put(&cd->kref, sr_kref_release);
-	scsi_device_put(cd->device);
+	scsi_device_put(sdev);
 	up(&sr_ref_sem);
 }
 
